lib/request.js

```diff
@@ -417,7 +417,13 @@ module.exports = {
 
   get subdomains() {
     var offset = this.app.subdomainOffset;
-    return (this.host || '')
+    var ip = /^(\d{1,3}\.){3}\d{1,3}$/gim;
```

Comment:
This doesn't accommodate IPv6. Also, I don't like the use of a regex here, is there a better way to do this?